src/pages/Account/NewCardForm.tsx

```diff
@@ -0,0 +1,294 @@
+import * as React from 'react';
+import { Button, Form, Row, Col, FormGroup, Label, FormFeedback, Input } from 'reactstrap';
+import { GET_PAYMENT_SOURCES, CREATE_PAYMENT_SOURCE } from 'networking/paymentSources';
+import { compose, graphql } from 'react-apollo';
+import { injectStripe, CardNumberElement, CardExpiryElement, CardCVCElement, PostalCodeElement } from 'react-stripe-elements';
+import StripeWrapper from 'HOCs/StripeWrapper';
+import styled from 'styled-components';
+
+interface Props {
+  stripe: Window['Stripe'];
+  createPaymentSource: (stripeToken: string) => void;
+  handleModal: (modal?: string) => void;
+  setAlert: (msg?: string) => void;
+}
+
+interface State {
+  completed: {
+    cardExpiry: boolean;
+    cardNumber: boolean;
+    cardCvc: boolean;
+    name: boolean;
+    postalCode: boolean;
+    [key: string]: boolean;
+  },
+  errors: {
+    cardExpiry: string;
+    cardNumber: string;
+    cardCvc: string;
+    name: string;
+    postalCode: string;
+    submit: string;
+    [key: string]: string;
+  };
+  isSubmitting: boolean;
+  name: string;
+}
+
+const StripeInputContainer = styled.div`
+  .__PrivateStripeElement {
+    width: 100%;
+  }
+`
+
+enum NewCardField {
+  CARD_NUMBER = 'cardNumber',
+  CARD_CVC = 'cardCvc',
+  NAME = 'name',
+  CARD_EXPIRY = 'cardExpiry',
+  POSTAL_CODE = 'postalCode',
+}
+
+class NewCardForm extends React.Component<Props, State> {
+  readonly state: State = {
+    completed: {
+      [NewCardField.CARD_EXPIRY]: false,
+      [NewCardField.CARD_NUMBER]: false,
+      [NewCardField.CARD_CVC]: false,
+      [NewCardField.NAME]: false,
+      [NewCardField.POSTAL_CODE]: false,
+    },
+    errors: {
+      [NewCardField.CARD_EXPIRY]: '',
+      [NewCardField.CARD_NUMBER]: '',
+      [NewCardField.CARD_CVC]: '',
+      [NewCardField.NAME]: '',
+      [NewCardField.POSTAL_CODE]: '',
```

Comment:
Think that `completed` and `errors` can be derived from `Object.keys(NewCardField)` to reduce duplication here

Reply:
thanks, updated!